src/update-feeds-loops/update-feeds-loops.ts

```diff
@@ -262,24 +269,32 @@ export const processBatch = async (
     blockNumber,
   });
   const {
-    config: { sponsorWalletMnemonic, chains, deviationThresholdCoefficient, walletDerivationScheme },
+    config: {
+      sponsorWalletMnemonic,
+      chains,
+      deviationThresholdCoefficient,
+      walletDerivationScheme,
+      signedApiUrls: configSignedApiBaseUrls,
+    },
   } = getState();
   const { contracts } = chains[chainId]!;
 
-  updateState((draft) => {
-    for (const dataFeed of batch) {
-      const receivedUrls = zip(dataFeed.signedApiUrls, dataFeed.beaconsWithData).map(([url, beacon]) => ({
-        url: `${url}/${beacon!.airnodeAddress}`,
-        airnodeAddress: beacon!.airnodeAddress,
-      }));
-      if (!draft.signedApiUrls) draft.signedApiUrls = {};
-      if (!draft.signedApiUrls[chainId]) draft.signedApiUrls[chainId] = {};
-      if (!draft.signedApiUrls[chainId]![providerName]) draft.signedApiUrls[chainId]![providerName] = {};
-      for (const { airnodeAddress, url } of receivedUrls) {
-        draft.signedApiUrls[chainId]![providerName]![airnodeAddress] = url;
-      }
-    }
-  });
+  // Generate signed API URLs for the batch
+  const signedApiUrls = batch
+    .map((dataFeed) =>
+      dataFeed.beaconsWithData.map((beacon, index) => {
+        const configSignedApiUrls = configSignedApiBaseUrls.map((baseUrl) => `${baseUrl}/${beacon.airnodeAddress}`);
+
+        // WARNING: contractSignedApiBaseUrl is an array of empty strings if it's not set on chain
```

Comment:
I think you mean `signedApiUrls` and not `contractSignedApiBaseUrl`.

But I think it's better to reflect this information early on. It would be better to change the empty strings to `null` [here](https://github.com/api3dao/airseeker-v2/blob/21674d8a2e7cab44a39ff53ebb56830231863e7b/src/update-feeds-loops/contracts.ts#L153).